### PR TITLE
Fix(DTFS2): fix vulnerable endpoints on external api

### PR DESCRIPTION
**Pull request by ChristianMcc**
## Introduction
Just casting to a typescript interface doesn't remove extra properties as at compile time the types are stripped away, leaving javascript passing on the whole req.body still. Added in object destructuring to remove any extra fields.

## Resolution
Add object destructuring to two endpoints in external API

### CODE REVIEW

File: external-api/api-tests/v1/eStore.api-test.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
+import { app } from '../../src/createApp';
```
Comment (ChristianMcc) [inline external-api/api-tests/v1/eStore.api-test.ts:1]: Mocking and testing in this file LGTM for testing the injection :)

File: external-api/src/v1/controllers/premium-schedule.controller.ts [hunk 2]
```diff
@@ -18,65 +18,66 @@ const headers: any = {
...
+const premiumScheduleCalls = {
+  /**
+   * Post premium schedule segments
+   * @param {Array} premiumSchedulePayload PS entries
+   * @returns Premium schedule segments
+   */
+  postPremiumSchedule: async (premiumSchedulePayload: any) => {
+    const premiumSchedulePayloadFormatted = premiumSchedulePayload;
+
+    if (objectIsEmpty(premiumSchedulePayload) || premiumSchedulePayload.facilityURN === 'PENDING') {
```
Comment (abhi-markan) [inline external-api/src/v1/controllers/premium-schedule.controller.ts:30]: Can `PENDING` please be picked from Constant.

File: external-api/src/v1/controllers/premium-schedule.controller.ts [hunk 3]
```diff
@@ -18,65 +18,66 @@
...
+    }
+    try {
+      const response = await axios({
+        method: 'post',
+        url: `${mdm}premium/schedule`,
+        headers,
+        data: [premiumSchedulePayloadFormatted],
+      }).catch((error: any) => {
+        console.error(
+          `Error calling POST Premium schedule with facilityURN: ${premiumSchedulePayloadFormatted.facilityURN} \n`,
```
[the remark this answers is not in the document]
Comment (abhi-markan) [reply external-api/src/v1/controllers/premium-schedule.controller.ts:47]: Can we please use `%s` specifier in the console.error.

```
console.error('Error calling POST Premium schedule with facilityURN %s:', premiumSchedulePayloadFormatted.facilityURN, error);
```

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Good work, few comments to address please.

File: external-api/api-tests/v1/eStore.api-test.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+// eslint-disable-next-line @typescript-eslint/no-var-requires
```
[the remark this answers is not in the document]
Comment (ChristianMcc) [reply external-api/api-tests/v1/eStore.api-test.ts:2]: This is used across every file in the external API, might be a bit of a bigger piece of work to remove this (couldn't figure out how to do it quickly)

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**